crates/cranelift/src/translate/code_translator.rs

```diff
@@ -2898,54 +2898,150 @@ pub fn translate_operator(
             // representation, so we don't actually need to do anything.
         }
 
-        Operator::ContNew { cont_type_index: _ } => {
-            // TODO(10248) This is added in a follow-up PR
-            return Err(wasmtime_environ::WasmError::Unsupported(
-                "codegen for stack switching instructions not implemented, yet".to_string(),
-            ));
+        Operator::ContNew { cont_type_index } => {
+            let cont_type_index = TypeIndex::from_u32(*cont_type_index);
+            let arg_types = environ.continuation_arguments(cont_type_index).to_vec();
+            let result_types = environ.continuation_returns(cont_type_index).to_vec();
```

Comment:
These should be able to be `SmallVec`s too.

Reply:
For args and returns, I guessed at a smallvec max of 8; I didn't see anywhere where we are bound to this, but smallvec will still fall back on heap allocations if we go beyond this.